contracts/Token.sol

```diff
@@ -131,6 +144,54 @@ contract Token is ERC2771Context, ERC20Permit, Pausable, AccessControl {
 
         // set requirements (can be 0 to allow everyone to send and receive tokens)
         requirements = _requirements;
+
+        // force-initialize the private variables name and symbol inherited from ERC20 using inline assembly
+        // solhint-disable-next-line no-inline-assembly
+        _storeString(3, _name);
+        _storeString(4, _symbol);
+    }
+
+    /**
+     * Taken from https://ethereum.stackexchange.com/questions/126269/how-to-store-and-retrieve-string-which-is-more-than-32-bytesor-could-be-less-th
+     */
+    function _storeString(uint256 _lengthSlot, string memory _newContent) private {
```

Comment:
If we could avoid that, that would be great.

Reply:
see above